streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -103,7 +103,6 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
                     true
                 )) {
                 outerJoinWindowStore = outerJoinWindowName.map(context::getStateStore);
-                sharedTimeTracker.nextTimeToEmit = context.currentSystemTimeMs();
```

Comment:
In `init()` the cached system time is still zero, so we need to delay the initialization to a later point.